### PR TITLE
Feature/actualizare conturi

### PR DESCRIPTION
**Pull request by MihAMi02**

### CODE REVIEW

File: src/main/java/org/example/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+        if(savedUser == null){
+            return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+        } else{
+            return new ResponseEntity<>(savedUser, HttpStatus.OK);
+        }
+    }
+
+    @GetMapping(value = "/getInfo")
+    public ResponseEntity<User> getUser(@RequestParam String email){
```
Comment (Piciorus) [inline src/main/java/org/example/controller/UserController.java:37]: this is not restful

File: src/main/java/org/example/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        User updatedUser = this.userService.getUserByEmail(email);
+
+        if(updatedUser == null){
+            return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+        } else{
+            return new ResponseEntity<>(updatedUser, HttpStatus.OK);
+        }
+    }
+
+    @PostMapping(value = "/updateInfo")
```
Comment (Piciorus) [inline src/main/java/org/example/controller/UserController.java]: this is not restful

File: src/main/java/org/example/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            return this.userRepository.save(user);
+        }
+
+        return null;
+    }
+
+    public User updateUser(User user){
+        if(validatePassword(user.getPassword())){
+            user.setPassword(passwordEncoder.encode(user.getPassword()));
+            EntityManagerFactory entityManagerFactory = Persistence.createEntityManagerFactory("CatalogPersistence");
```
Comment (Piciorus) [inline src/main/java/org/example/service/UserService.java]: do we need this for an update user?

File: src/main/java/org/example/service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            EntityManager entityManager = entityManagerFactory.createEntityManager();
+
+            User temp = this.getUserByEmail(user.getEmail());
+            if(temp != null){
+                // Begin a transaction
+                EntityTransaction transaction = entityManager.getTransaction();
```
Comment (Piciorus) [inline src/main/java/org/example/service/UserService.java]: why we need this?

File: src/main/java/org/example/service/UserService.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+    public boolean validateNames(String name) {
+        if (name == null)
+            return false;
+        Pattern pattern = Pattern.compile("^[A-Za-z]+(?:[ -][A-Za-z]+)*$");
+        Matcher matcher = pattern.matcher(name);
+        return matcher.find();
+    }
+
+
```
Comment (Piciorus) [inline src/main/java/org/example/service/UserService.java]: format file

File: src/main/java/org/example/test/controller/UserControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Test
+    void updateUser_ValidUser_ReturnsUpdatedUser() {
+        // Mocking
+        User user = new User(); // Create a valid user
+        when(userService.updateUser(user)).thenReturn(user);
+
+        // Test
+        //ResponseEntity<User> response = userController.updateUser(user.getLastname(), user.getFirstname(), user.getEmail(), user.getPassword(), user.getRole().toString());
+
```
Comment (Piciorus) [inline src/main/java/org/example/test/controller/UserControllerTest.java]: comment code

File: src/main/resources/META-INF/persistence.xml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    <persistence-unit name="CatalogPersistence">
+        <provider>org.hibernate.jpa.HibernatePersistenceProvider</provider>
+
+        <!-- Specify the JPA entity classes -->
+        <class>org.example.model.User</class>
+
+        <properties>
+            <property name="hibernate.dialect" value="org.hibernate.dialect.MySQLDialect"/>
+            <property name="hibernate.hbm2ddl.auto" value="update"/>
+            <property name="javax.persistence.jdbc.driver" value="com.mysql.cj.jdbc.Driver"/>
```
Comment (Piciorus) [inline src/main/resources/META-INF/persistence.xml]: we have in properties those

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -0,0 +1,9 @@
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/Catalog
+spring.datasource.username=root
+spring.datasource.password=12345678
```
Comment (Piciorus) [inline src/main/resources/application.properties]: can you sync with dev and to try to have the same credentials?

File: src/main/resources/static/Test.html [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        var formData = {
+          lastname: $("#lastname").val(),
+          firstname: $("#firstname").val(),
+          email: $("#email").val(),
+          password: $("#password").val(),
+          role: $("#role").val()
+        };
+
+        // Log the form data (you can do any manipulation here)
+        console.log(formData);
```
Comment (Piciorus) [inline src/main/resources/static/Test.html]: remove all console log

File: src/main/resources/static/Test.html [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+<html lang="en">
+<head>
+  <title>Test thingy</title>
+  <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/htmx.org/dist/htmx.min.js"></script>
+  <script type="text/javascript" src="https://cdn.jsdelivr.net/npm/hyperscript.org/dist/_hyperscript.js"></script>
+  <script src="https://code.jquery.com/jquery-3.6.4.min.js"></script>
+  <script>
+    $(document).ready(function() {
+      $("#registrationForm").submit(function(event) {
+        // Prevent the default form submission
```
Comment (Piciorus) [inline src/main/resources/static/Test.html]: can you move this logic in another file and call it here

**Review by Piciorus — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/controller/UserController.java (hunk 1 above)
Comment (MihAMi02) [reply src/main/java/org/example/controller/UserController.java:37]: What would be restful?

File: src/main/java/org/example/service/UserService.java (hunk 3 above)
Comment (MihAMi02) [reply src/main/java/org/example/service/UserService.java]: Afaik that's the only way to update a db entry, also requiring persistence.xml

File: src/main/resources/static/Test.html (hunk 9 above)
Comment (MihAMi02) [reply src/main/resources/static/Test.html]: Test.html was commited by mistake and is now removed

File: src/main/java/org/example/controller/UserController.java (hunk 1 above)
Comment (Piciorus) [reply src/main/java/org/example/controller/UserController.java:37]: https://www.sitepoint.com/rest-api/ see here

Comment (Piciorus) [reply src/main/java/org/example/controller/UserController.java:37]: for a get you ll have /user in path not /getInfo

File: src/main/java/org/example/service/UserService.java (hunk 3 above)
Comment (Piciorus) [reply src/main/java/org/example/service/UserService.java]: we have jpa repository with methods for save delete findById..you can perform an update with those . The repos interacts with db

**Comment by Piciorus** [comment]
please resolve conflicts

**Review by DragosPetri — DISMISSED** [review]

**Review by Piciorus — DISMISSED** [review]

**Review by VladP2002 — DISMISSED** [review]

**Review by DragosPetri — DISMISSED** [review]

**Pull request merged**